### PR TITLE
8252722: More Swing plaf APIs that rely on default constructors

### PR DESCRIPTION
**Pull request by prsadhuk**
Please review a fix for issue where it was seen that several classes in plaf package rely on default constructors as part of their public API.

It's to be noted that "A no-arg public constructor is generated by the compiler for a class if it does not declare an
explicit constructor. While convenient, this is inappropriate for many kinds of formal classes, both because the
constructor will have no javadoc and because the constructor may be unintended."

For the JDK, classes intended to be used outside of the JDK, public classes in exported packages, should not rely on
default constructors.

Proposed fix is to add explicit public no-arg constructors for public classes and protected no-arg constructor for
protected classes for javax.swing module

---------
### Progress
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue
- [x] Change must be properly reviewed

### Issue
 * [JDK-8252722](https://bugs.openjdk.java.net/browse/JDK-8252722): More Swing plaf APIs that rely on default constructors

### Reviewers
 * [Phil Race](https://openjdk.java.net/census#prr) (@prrace - **Reviewer**) ⚠️ Review applies to 2b6273bfbb8d26d8b66b12e93e41a93f7708d059

### Download
`$ git fetch https://git.openjdk.java.net/jdk pull/251/head:pull/251`
`$ git checkout pull/251`

### CODE REVIEW

**Comment by prsadhuk** [comment]
CSR: https://bugs.openjdk.java.net/browse/JDK-8253358

**Comment by prsadhuk** [comment]
Any reviewers?

**Review by prrace — APPROVED** [review]

File: src/java.desktop/share/classes/javax/swing/plaf/basic/BasicComboPopup.java [hunk 1]
```diff
@@ -733,6 +733,11 @@ public void actionPerformed(ActionEvent e){
      * @see BasicComboPopup#createMouseListener
      */
     protected class InvocationMouseHandler extends MouseAdapter {
+        /**
+         * Constructor for subclasses to call.
```
[the remark this answers is not in the document]
Comment (prsadhuk) [reply src/java.desktop/share/classes/javax/swing/plaf/basic/BasicComboPopup.java]: it's outdated file you are reviewing..It's already changed yesterday.

[the remark this answers is not in the document]
Comment (prrace) [reply src/java.desktop/share/classes/javax/swing/plaf/basic/BasicComboPopup.java]: That is weird. I wrote that comment 2 days ago!

[the remark this answers is not in the document]
Comment (prrace) [reply src/java.desktop/share/classes/javax/swing/plaf/basic/BasicComboPopup.java]: I also wrote the comments about a->an at least a day ago .. and github also claims they were made just 34 minutes ago. Very strange.

**Pull request closed without merging**